### PR TITLE
New telugu cheat sheet

### PR DESCRIPTION
**Pull request by rahulroxx**
a

IA : https://duck.co/ia/view/telugu_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@rahulroxx Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Telugu Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/telugu_language_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/telugu.json [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    "name": "Telegu",
+    "description": "Basic telegu words and phrases",
+
+    "metadata": {
+       "sourceName" : "Wikitravel",
+      "sourceUrl" : "http://wikitravel.org/en/Telugu_phrasebook"
+     },
+
+"aliases": [
+       "telegu phrases", "english to telegu", "basic telegu phrases", "basic telegu","useful telegu phrases"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/telugu.json]: Get rid of the aliases that have `phrases` in them.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Telugu Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/telugu_language_cheat_sheet_mobile.png)

**Comment by rahulroxx** [comment]
@GuiltyDolphin  "Removed phrases"

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Telugu Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/telugu_language_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/telugu.json [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    "id": "telugu_cheat_sheet",
+    "name": "Telegu",
+    "description": "Basic telegu words and phrases",
+
+    "metadata": {
+       "sourceName" : "Wikitravel",
+      "sourceUrl" : "http://wikitravel.org/en/Telugu_phrasebook"
+     },
+
+"aliases": [
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/telugu.json]: Indenting is a bit off here.

Make sure you are using 4 spaces per indent ***throughout the file***.

**Comment by rahulroxx** [comment]
@GuiltyDolphin  "Indentation Fixed"

**Comment by rahulroxx** [comment]
@GuiltyDolphin "Fixed indents"

**Comment by rahulroxx** [comment]
@GuiltyDolphin telugu.json moved from json/telugu.json to language/

File: share/goodie/cheat_sheets/json/language/telugu.json [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+    "description":"Basic telegu words and phrases",
+    "metadata":{  
+        "sourceName":"Wikitravel",
+        "sourceUrl":"http://wikitravel.org/en/Telugu_phrasebook"
+    },
+    "aliases":[  
+        "english to telegu",
+        "basic telegu"
+    ],
+    "template_type":"language",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/language/telugu.json]: Lets get some spacing going! There should be a space between the colon and value, e.g., `key: value`.

**Comment by GuiltyDolphin** [comment]
@rahulroxx It looks like you've copied it, rather than moved it. Now you have two files! Might want to get that sorted.

**Comment by rahulroxx** [comment]
@GuiltyDolphin  I moved it !

File: share/goodie/cheat_sheets/json/language/telugu.json [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+                "val":"Tommidi"
+            },
+            {  
+                "key":"10",
+                "val":"Padi"
+            }
+        ],
+        "Problems":[  
+            {  
+                "key":"Don't touch me! ",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/language/telugu.json]: Get rid of the space after the `!`, there's a couple of cases of this elsewhere too.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Telugu Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/telugu_language_cheat_sheet_mobile.png)

**Comment by rahulroxx** [comment]
@GuiltyDolphin  Removed unwanted spaces !

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Telugu Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/telugu_language_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/language/telugu.json [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+                "val":"మీ పేరేమండి ?",
+                "trn":"Mee paeraemaṅdi?"
+            },
+            {  
+                "key":"Nice to meet you",
+                "val":"Mimmalni kalisi nanduku santhosam ga unnadi."
+            },
+            {  
+                "key":"Look out!/Look over there!",
+                "val":"Atu choodu! "
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/language/telugu.json]: Looks like more spaces after punctuation, I'd suggest just having a skim through the keys and values to make sure there aren't any more (or use `grep`, if you're feeling adventurous).

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Telugu Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/telugu_language_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Telugu Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/telugu_language_cheat_sheet_mobile.png)

**Comment by rahulroxx** [comment]
@GuiltyDolphin i removed remaining spaces

**Comment by rahulroxx** [comment]
@GuiltyDolphin i can't see telugu.json in upstream master ?

**Comment by rahulroxx** [comment]
@GuiltyDolphin I mean ,  will there be a merge for this PR ?

File: share/goodie/cheat_sheets/json/language/telugu.json [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+                "trn":"Naenu baagunnaanu, meeru aelaa unnaaru?"
+            },
+            {  
+                "key":"What is your name?",
+                "val":"మీ పేరేమండి?",
+                "trn":"Mee paeraemaṅdi?"
+            },
+            {  
+                "key":"Nice to meet you",
+                "val":"Mimmalni kalisi nanduku santhosam ga unnadi."
```
Comment (mohan08p) [inline share/goodie/cheat_sheets/json/language/telugu.json]: Get rid of periods... And, I think.. that is trn field which you kept as val fields..  Kindly, make changes to those fields and add appropriates..

File: share/goodie/cheat_sheets/json/language/telugu.json [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+            },
+            {  
+                "key":"Look out!/Look over there!",
+                "val":"Atu choodu!"
```
Comment (mohan08p) [inline share/goodie/cheat_sheets/json/language/telugu.json]: Same here... First val fields must be your cheat sheet named for.. .i.e. Telugu respective to the key field. Also, check the subsequent fields...

File: share/goodie/cheat_sheets/json/language/telugu.json [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+            },
+            {  
+                "key":"I've been injured.",
+                "val":"Naaku debba tagilindi"
+            }
+        ],
+        "Time":[  
+            {  
+                "key":"Morning",
+                "value":"Udayam",
```
Comment (mohan08p) [inline share/goodie/cheat_sheets/json/language/telugu.json]: fields been reversed as the val should contain native value and trn should its translation into English, respectively.

File: share/goodie/cheat_sheets/json/language/telugu.json [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+                "trn":"ఉదయం"
+            },
+            {  
+                "key":"Before",
+                "value":"Mundu",
+                "trb":"ముందు"
```
Comment (mohan08p) [inline share/goodie/cheat_sheets/json/language/telugu.json]: check spelling of "trn" ...

**Comment by rahulroxx** [comment]
@mohan08p  i'll get rid of the periods . Could you please explain the last comment

**Comment by rahulroxx** [comment]
@mohan08p i'll get rid of the periods , could you please explain the last comment

**Comment by mohan08p** [comment]
Kindly, refer this marathi cheat sheet, https://github.com/duckduckgo/zeroclickinfo-goodies/pull/2173/files
you will understand. Also, connect on the slack channel.. so you will better understands from community members.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Telugu Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/telugu_language_cheat_sheet_mobile.png)

**Comment by rahulroxx** [comment]
@mohan08p Thanks ! I fixed everything you said

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Telugu Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/telugu_language_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/language/telugu.json [hunk 10]
```diff
@@ -0,0 +1,278 @@
...
+                "key":"Evening",
+                "val":" సాయంత్రం",
+                "trn":"Sayantram"
+                
+            },
+            {  
+                "key":"Night",
+                "val":"రాత్రి",
+                "trn":"Ratri"
+
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/language/telugu.json]: Don't think we need this extra line.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Telugu Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/telugu_language_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
@rahulroxx thanks for this! Are you a native Telugu speaker? If not I'd like to get some feedback from the community to make sure the translations are correct :+1:

**Comment by rahulroxx** [comment]
@moollaza i am not a native telegu speaker ! but i learnt from my family .

File: share/goodie/cheat_sheets/json/language/telugu.json [hunk 11]
```diff
@@ -0,0 +1,272 @@
+{  
+    "id":"telugu_cheat_sheet",
+    "name":"Telegu",
```
Comment (pnpninja) [inline share/goodie/cheat_sheets/json/language/telugu.json]: Change name to "Telugu"

**Comment by rahulroxx** [comment]
I think we should merge PR i am pretty confident all the information is correct ! @moollaza

**Comment by moollaza** [comment]
Thanks! This LGTM :+1:

**Comment by rahulroxx** [comment]
Thanks :)

**Comment by tagawa** [comment]
@rahulroxx, thanks again for creating the Telugu cheat sheet! You’ve done a really good job here and we'd like you to help make DuckDuckGo even better for yourself and others interested in languages.

When you're ready to move on, please:

- Create a new Instant Answer for searches you do related to languages

OR

- Test live languages Instant Answers (or those in development) for bugs and suggest improvements

Detailed instructions explaining how you can help are available at the top of this [page](https://github.com/duckduckgo/duckduckgo/wiki/Creating-or-Improving-Instant-Answers).

Thanks again!

**Pull request merged**